content/actions/reference/environment-variables.md

```diff
@@ -72,6 +72,9 @@ We strongly recommend that actions use environment variables to access the files
 | `GITHUB_SERVER_URL`| Returns the URL of the {% data variables.product.product_name %} server. For example: `https://{% data variables.product.product_url %}`.
 | `GITHUB_API_URL` | Returns the API URL. For example: `{% data variables.product.api_url_code %}`.
 | `GITHUB_GRAPHQL_URL` | Returns the GraphQL API URL. For example: `{% data variables.product.graphql_url_code %}`.
+| `RUNNER_OS` | The operating system of the runner executing the job. Possible values are `Linux`, `Windows`, or `macOS`.
+| `RUNNER_TEMP` | The path to a temporary directory on the runner. This directory is emptied after each job completes.
```

Comment:
Is this behaviour guaranteed, or might there still be files remaining if a self-hosted runner has previously terminated unexpectedly?

Reply:
And just to confirm: is the directory cleared before the job starts, or after the job ends (whether success or failure)?

Reply:
It seemed obvious so I wasn't explicit about this in the commit message, but this proposed text was copied from the corresponding entries in _context-and-expression-syntax-for-github-actions.md_. So you might apply any wording improvements there too.

Reply:
The proposed rewording changes a definitive statement (“guaranteed to be empty at [job start]”) to a vague one (“emptied after each job completes” — What exactly does ”complete” mean, in the context of failures? Could anything else be putting files there in between jobs?). In fact for users of GH Actions it doesn't matter exactly when the directory is cleared, because the difference is not observable. The only thing that is observable is whether the the directory is empty at the start of each job, so IMHO the existing text is the most useful for readers. (I suppose some readers may also care that temporary files are not left lying around afterwards either, but sure-to-be-empty-on-start is the critical part for most.)

Looking at [_JobRunner.cs_](https://github.com/actions/runner/blob/main/src/Runner.Worker/JobRunner.cs) (search for `_tempDirectoryManager`), you can see that the current implementation clears the directory both before **and** after each job.

Reply:
`RUNNER_TEMP` will be cleaned up by the runner in the best effort way. 
the runner will try to clean it up before the job and also after the job, so even you unplug the power from the runner to cause the post-job clean not to run, it will still get cleaned before the next job starts running.
However, the runner can't delete files if the file is created by a different user and the runner user account don't have permission to delete the file.

Reply:
Thanks @TingluoHuang - I've updated the description to mention this possibility: https://github.com/github/docs/pull/6740/commits/031ae8476c5aca6d57c6463b75c9eeb6173084a1